torch/distributed/utils.py

```diff
@@ -337,3 +338,11 @@ def _replace_by_prefix(
 
 def _data_ptr_allocated(tensor: torch.Tensor) -> bool:
     return tensor.untyped_storage().data_ptr() > 0
+
+
+class _DeviceModule(Protocol):  # noqa: PYI046
```

Comment:
Are you using the protocol here to represent a python module (e.g. torch.cuda is a module that defines these methods)?  It makes sense I think, but I've never seen this technique, is it a recommended/pythonic way to do things?

Reply:
Yes, protocol here is used for device module static check, as other devices may not inherited torch.cuda gernerally. I think it is a formal way in type hinting (PEP484 and PEP544)